page/Help.js

```diff
@@ -0,0 +1,331 @@
+    remnants: require("../assets/help/remnants-24px_default.png")
+}
+
+const HelpAudio = {
+    home: require("../assets/audio/sound.mp3")
+}
+
+const PlayingIcon = require("../assets/help/record_voice_over_playing-24px_default.png");
+const ListenIcon = require("../assets/help/audio_help-24px_default.png");
+
+function withSound(name) {
+    return new Promise(function(resolve, reject) {
+        let sound = new Sound(name, (error) => {
+            if (error) {
+                reject(sound, error);
+            } else {
+                resolve(sound);
+            }
+        })
+    });
+}
+
+
+export class SectionedScroller extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            pageHeight: 500,
+            soundKey: null,
+            soundPlaying: null,
+            soundLoading: null
+        };
+    }
+
+    _scrollTo(name, animated=false) {
+        let child = this.refs[name];
+
+        if (child) {
+            let nodeHandle = ReactNative.findNodeHandle(this._scroller);
+            child.measureLayout(nodeHandle, (_x, y) => {
+                this._scroller.scrollTo({x: 0, y: y-30, animated: animated});
+            }, (error) => {
+                console.log(error);
+            })
+        }
+    }
+
+    stopSound() {
+        let {soundPlaying} = this.state;
+
+        if (!soundPlaying) return;
+
+        soundPlaying.stop();
+        this.setState({
+            soundKey: null,
+            soundPlaying: null
+        });
+    }
+
+    onPlayingComplete(success, name, sound) {
+        if (this.state.soundKey === name) {
+            this.setState({
+                soundKey: null,
+                soundPlaying: null
+            });
+        }
+    }
+
+    playSound(key) {
+        console.log(`Play sound: ${key}`);
+        let {state} = this;
+
+        if (state.soundPlaying) {
+            this.stopSound();
+
+            if (state.soundKey === key)
+                return;
+        }
+
+        this.setState({ soundKey: key, soundLoading: true });
+        withSound(HelpAudio.home)
+            .then((sound) => {
+                console.log(sound);
+                if (this.state.soundKey !== key) return;
+
+                console.log("boop");
+                sound.play((success) => {
+                    this.onPlayingComplete(success, key, sound)
+                });
+                this.setState({
+                    soundPlaying: sound
+                })
+            })
+            .finally(() => {
+                if (this.state.soundKey === key)
+                    this.setState({ soundLoading: false });
+            })
+    }
+
+    onLayout = (e) => {
+        this.setState({ pageHeight: e.nativeEvent.layout.height });
+
+        if (this.props.selected)
+            this._scrollTo(this.props.selected);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.selected && prevProps !== this.props.selected)
```

Comment:
Shouldn't it be prevProps.selected !== this.props.selected?

Reply:
Yes. Fixed.